pkg/cache/cache.go

```diff
@@ -688,6 +690,52 @@ func (c *Cache) Usage(cqObj *kueue.ClusterQueue) (*ClusterQueueUsageStats, error
 	return stats, nil
 }
 
+type CohortUsageStats struct {
+	WeightedShare int64
+}
+
+func (c *Cache) CohortUsage(cohortObj *kueuealpha.Cohort) (*CohortUsageStats, error) {
+	c.RLock()
+	defer c.RUnlock()
+
+	cohort := c.hm.Cohort(kueue.CohortReference(cohortObj.Name))
+	if cohort == nil {
+		return nil, ErrCohortNotFound
+	}
+
+	stats := &CohortUsageStats{}
+	if c.fairSharingEnabled {
+		weightedShare, _ := dominantResourceShare(cohort, nil)
+		stats.WeightedShare = int64(weightedShare)
+	}
+
+	return stats, nil
+}
+
+func (c *Cache) CohortAncestors(cohortName kueue.CohortReference) []kueue.CohortReference {
```

Comment:
this should also return the Cohort it was called with. the interface/name may be a little clearer if we call it ClusterQueueAncestors, and call it with a CQ?

Reply:
> this should also return the Cohort it was called with.

Not sure I got why we need to return current cohort?

Reply:
ahh, well in the callsite you can get rid of reconcile request for cq.Cohort (which could be empty string anyway), and just process all parent Cohorts of the CQ in one loop?

Reply:
In that case we need to change name. Something like CohortAndAncestors(). WDYT?

Reply:
> the interface/name may be a little clearer if we call it ClusterQueueAncestors, and call it with a CQ?

what about this?

Reply:
Ah, I got what you mean. Thanks for explanation!